kfs-core/src/main/java/edu/arizona/kfs/pdp/service/impl/PayeeAchAccountServiceImpl.java

```diff
@@ -0,0 +1,268 @@
+		
+		if (payee != null) {
+			inGroupFlag = getGroupService().isMemberOfGroup(payee.getPrincipalId(), groupId);
+		}
+		
+		return inGroupFlag;
+	}
+
+	@Override
+	public List<String> getPrincipalIdsInOverrideGroup() {
+		String groupId = parameterService.getParameterValueAsString(LoadAchPayeeBankAcctStep.class, KFSParameterKeyConstants.ACHFileConstants.BANKING_INFORMATION_OVERRIDE_GROUP);
+		if (StringUtils.isBlank(groupId)) {
+			return null;
+		}
+		
+		List<String> overrideIds = getGroupService().getDirectMemberPrincipalIds(groupId);
+		return overrideIds;
+	}
+	
+	protected List<String> addEmplIdsToPrincipalIds(List<String> principalIds) {
+		if (principalIds == null) {
+			return null;
+		}
+		
+		List<String> emplIds = new ArrayList<String>();
+		
+		Person person;
+		String emplId;
+		for (String principalId : principalIds) {
+			person = personService.getPerson(principalId);
+			emplId = person.getEmployeeId();
+			if (StringUtils.isNotBlank(emplId)) {
+				if (!emplIds.contains(emplId)) {
+					emplIds.add(emplId);
+				}
+			}
+		}
+		
+		emplIds.addAll(principalIds);
+		return emplIds;
+	}
+	
+	protected Integer resetRecordsOfType(String inputFileName) {
+		Integer recordsReset = new Integer(0);
+		
+		String payeeType = validateTypeUsedInFile(inputFileName);
+		if (StringUtils.isBlank(payeeType)) {
+			throw new RuntimeException("PayeeType was blank from file [" + inputFileName + "]. This should never happen.");
+		}
+		
+		List<String> overridePrincipalIds = getPrincipalIdsInOverrideGroup();
+		
+		List<String> overridePrincipalAndEmplIds = addEmplIdsToPrincipalIds(overridePrincipalIds);
+		
+		recordsReset = removePayeeAchAccountsByPayeeType(payeeType, overridePrincipalAndEmplIds);
+		
+		return recordsReset;
+	}
+	
+	// this method will remove those accounts that match the PayeeIdentifierType on the File provided but not the list of payeeIdNumber's found on the BANKING_INFORMATION_OVERRIDE_GROUP group
+	private Integer removePayeeAchAccountsByPayeeType(String payeeType, List<String> overridePrincipalAndEmplIds) {
+		Integer count;
+		Map<String, String> criteria = new HashMap<String, String>();
+		criteria.put(PdpPropertyConstants.PAYEE_IDENTIFIER_TYPE_CODE, payeeType);
+		
+		Collection<PayeeACHAccount> payees = businessObjectService.findMatching(PayeeACHAccount.class, criteria);
+		List<PayeeACHAccount> payees2 = new ArrayList<PayeeACHAccount>();
+		
+		if (overridePrincipalAndEmplIds != null && !overridePrincipalAndEmplIds.isEmpty()) {
+			for (PayeeACHAccount payee : payees) {
+				if (!overridePrincipalAndEmplIds.contains(payee.getPayeeIdNumber())) {
+					payees2.add(payee);
+				}
+			}
+		}
+		else {
+			count = payees.size();
+			
+			for (PayeeACHAccount payee : payees) {
+				LOG.debug("deleted payee = " + payee);
+				businessObjectService.delete(payee);
+			}
+			
+			return count;
+		}
+		
+		count = payees2.size();
```

Comment:
Ah yes, I wondered if I had the logic as I was trying to fall asleep last night (code runs free then haha). I have sent an email with an example snippet of code I'd like you to peek at, with further info for your perusal (hint: it will be your discretion where to go from there).